api doc with notion (#16)
apitoolkit/apitoolkit-landing: content/blog/Simple-Steps-to-Using-Notion-for-API-Documentation/index.md

---
title: "Simple Steps to Using Notion for API Documentation"
date: 2022-11-01T08:05:56+02:00
author: jessica
categories: 
- APIs
---

![Notion logo](./notion-logo.png)

We talked about using [Notion for documentation](../using-notion-for-API-documentation/index.md) last week; today, we'll take a quick look at how to use the same software to pen down our API documentation. Note that there is a slight difference between documentation and API documentation and not all tools can be used to create an API doc.

> APIs have increased significantly with the use of API documentation. With proper API documentation, users have a better experience with the functionality and instructions for using and integrating the project, coupled with updates on the API lifespan. This doesn’t entirely mean that you can’t use an API without documentation but the idea is to make things a lot easier for the user who might probably be a first time user and give access to the technical content of the API and integration rules to make work easier and convienient for them. There are lots of tools that can help simplify and manage this document and notion is one of those tools, Notion is an all-in-one workspace, it goes beyond just basic documentation to API documentation giving you access to code blocks, navigation bars, etc.

Reviewer: What do you mean when you say APIs have increased? To the mest of my knowlege, it's not possible to an api to increase. The number of endpoints in an API could increase for instance. Maybe you want to say that we have more APIs being built than ever before ?

## Using Notion For API Doc
Creating an API Doc can be quite complex, especially at the point of finding a template that works. Today we will create an API doc with an existing template and also create one from scratch just incase you dont find a template that works for you.

### Working with a Template
We'll make use of a template from [Beestat](https://www.notion.so/API-Documentation-e4a7746e6a3f45dbb58ea6b45b8f9744). This template is a perfect choice for API documentation. It includes a navigation bar at the top page, a sub-page, lists, and code blocks to generate all of the product documentation in Notion. It is divided into three sections: the navigation section, the documentation section, and the reference section.

![Navigation bar](./navigation.png)
**The Navigation:** The navigation section is also what we know as the table of content section. It is created to automatically update as you add new header sections (i.e /h1, /h2, /h3) to your page. This is a really practical and simply one more fantastic feature Notion has because it works like a link that takes you right to the header you clicked on.

![Documentation](./content.png)

> **The Documentation:** This part contains the content that you want your readers to read. Here you find the code blocks, the images, the videos and all that makes your doc what it ought to be. There is no word limit, it can be anything, it all depends on you.

Reviewer: Instead of there is no word limit, mention that consise is usually better

![Reference](./resource.png)

> **The Reference:** The reference is a list of resources located at the page's footer. You can add additional pages here, and then use the notio pagename to refer to them elsewhere in the documentation. This is incredibly practical because it allows for numerous references to the same page and provides a simple "back" button on those subpages.

Reviewer: ```suggestion
**The Reference:** The reference is a list of resources located at the page's footer. You can add additional pages here, and then use the notion page name to refer to them elsewhere in the documentation. This is incredibly practical because it allows for numerous references to the same page and provides a simple "back" button on those subpages.
```

---
title: "Simple Steps to Using Notion for API Documentation"
date: 2022-11-01T08:05:56+02:00
author: jessica
categories: 
- APIs
---

![Notion logo](./notion-logo.png)

We talked about using [Notion for documentation](../using-notion-for-API-documentation/index.md) last week; today, we'll take a quick look at how to use the same software to pen down our API documentation. Note that there is a slight difference between documentation and API documentation and not all tools can be used to create an API doc.

> The use of APIs by companies have increased significantly with the use of API documentation. With proper API documentation, users have a better experience with the functionality and instructions for using and integrating the project, coupled with updates on the API lifespan. This doesn’t entirely mean that you can’t use an API without documentation but the idea is to make things a lot easier for the user who might probably be a first time user and give access to the technical content of the API and integration rules to make work easier and convienient for them. There are lots of tools that can help simplify and manage this document and notion is one of those tools, Notion is an all-in-one workspace, it goes beyond just basic documentation to API documentation giving you access to code blocks, navigation bars, etc.

Reviewer: `The use of APIs by companies have increased significantly with the use of API documentation.` This statement is not true.

> ## Using Notion For API Doc

Reviewer: ```suggestion
## Using Notion For API Documentation
```

> Creating an API Doc can be quite complex, especially at the point of finding a template that works. Today we will create an API doc with an existing template and also create one from scratch  which will cover all thee important components of APIs like endpoints, the fields that makes up the enpoints resourses and other parameters. Sometimes the template dont include all of this parameters so we can decide to create one from scratch.

Reviewer: ```suggestion
Creating API Documentation can be quite complex, especially at the point of finding a template that works. Today we will create an API doc with an existing template and also create one from scratch  which will cover all thee important components of APIs like endpoints, the fields that makes up the enpoints resourses and other parameters. Sometimes the template dont include all of this parameters so we can decide to create one from scratch.
```

<video controls src="creating-an-api-doc-with-notion.mp4" ></video>

This video teaches how to sample request and responses for every endpoint on the app. Request and Response are core components of APIs. Request is the action that occurs when a call is made through the server by adding a URL endpoint. Anytime a call is made to the server using an API, this counts as an API request. Operations like saves, signup, logins, and  queries, among others are examples of request. The very moment an endpoint is added to a URL or you even register on an app or website, you will be making an API request and the result thereafter is the Response.
<video controls src="request-and-response" ></video>

This video guides you through the reference/resource adding process. This covers for the information that is returned by the API. Its generally modelled as a collection with tags where resources are put together. There are different categories of resources though. 
<video controls src="objectresource" ></video>

### Working with a Template

> We'll make use of a template from [Beestat](https://www.notion.so/API-Documentation-e4a7746e6a3f45dbb58ea6b45b8f9744). This template is a perfect choice for API documentation. It includes a navigation bar at the top page, a sub-page, lists, and code blocks to generate all of the product documentation in Notion. It is divided into three sections: the navigation section, the documentation section, and the reference section.

Reviewer: ```suggestion
We'll make use of a template from [Beestat](https://www.notion.so/API-Documentation-e4a7746e6a3f45dbb58ea6b45b8f9744). This template is a good choice for API documentation. It includes a navigation bar at the top page, a sub-page, lists, and code blocks to generate all of the product documentation in Notion. It is divided into three sections: the navigation section, the documentation section, and the reference section.
```
